sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -787,6 +788,10 @@ abstract class ToTimestamp
               formatter.parse(
```

Comment:
can we create a util function to do it? e.g.
```
def parseTimestamp(formatter: TimestampFormatter,  str: String, result: LongWrapper): Boolean = {
  try {
    ...
    result.set(...)
    true
  } catch {
    case e: DateTimeParseException =>
      if (FAIL) throw ...
      else false
    case NonFatal(_) => false
  }
}
```

and the caller side
```
val result = new LongWrapper
if (parseTimestamp(formatter, str, result)) {
  result.get ... // whatever operations
} else {
  null
}
```

Reply:
and do we need to fix more places? `formatter.parse` is being used in many places IIRC.

Reply:
> can we create a util function to do it?

The function demo you provided is clearer, but since we need to change both sides of code generation and normal mode, the current abstract might be better?

> and do we need to fix more places?

Actually, in this PR, we tracked all the place which used the original config `LEGACY_TIME_PARSER_ENABLED ` and do the fix in UT. For every case that `legacy` mode have a different result with `corrected` mode, I added an explicit check in the UT. (The only exception is https://github.com/apache/spark/pull/27537/files#diff-ab881d9b6bceac0a14656dee940917f4R720, which use different format pattern in legacy and corrected mode, so I think this shouldn't be the cases we want to fix here.)
cc @MaxGekk for double-checking.